### PR TITLE
SQL: Enhance message for PERCENTILE[_RANK] with field as 2nd arg

### PR DESCRIPTION
**Pull request by matriv**
Enhance error message for the case that the 2nd argument of PERCENTILE
and PERCENTILE_RANK is not a foldable, as it doesn't make sense to have
a dynamic value coming from a field.

Fixes: #36903

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java [hunk 1]
```diff
@@ -61,6 +62,10 @@ public DataType dataType() {
 
     @Override
     public String innerName() {
+        if (!percent.foldable()) {
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java]: The check happens too late when the Percentile is going to be executed. It's better to move it up-front in the `resolveType` since that's where we typically check the arguments.
This would also avoid having to create an exception and instead return a resolution message.

**Review by costin — CHANGES_REQUESTED** [review]
The checks needs moving at resolution time (instead of in a getter).

**Comment by matriv** [comment]
@costin Thanks for the comment, Fixup pushed.

**Comment by matriv** [comment]
run the gradle build tests 2

**Review by astefan — APPROVED** [review]
LGTM. Just a suggestion: does the documentation page need an improvement or is it enough to mention the second parameter as "numeric expression" (as it is now)?

**Comment by matriv** [comment]
@asterfan Thanks for the comment, added a more detailed explanation there.

**Comment by matriv** [comment]
@astefan Thanks for the comment, added a more detailed explanation there.

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java [hunk 2]
```diff
@@ -41,13 +42,17 @@ public Percentile replaceChildren(List<Expression> newChildren) {
 
     @Override
     protected TypeResolution resolveType() {
-        TypeResolution resolution = super.resolveType();
+        if (!percent.foldable()) {
+            throw new SqlIllegalArgumentException("2nd argument: [{}] of PERCENTILE cannot be based on " +
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java]: I would rephrase this to something shorter such as "must be a constant, received [{}]".

**Review by costin — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
Backported to `6.x` with 685d654fbb9ed6fc92b494f1aa20baa10f80e1da
